datalad/metadata/search.py

```diff
@@ -124,6 +124,8 @@ def __call__(match,
                   match, dataset)
         try:
             ds = require_dataset(dataset, check_installed=True, purpose='dataset search')
+            if ds.id is None:
+                raise NoDatasetArgumentFound
```

Comment:
I would have added a message that repository is there but it is not a dataset since missing an ID